src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py

```diff
@@ -172,114 +204,346 @@ def visit_StencilClosure(
+        array_mapping = {**input_mapping, **conn_mapping}
+        symbol_mapping = map_nested_sdfg_symbols(closure_sdfg, nsdfg, array_mapping)
 
         nsdfg_node, map_entry, map_exit = self._add_mapped_nested_sdfg(
             closure_state,
-            sdfg=context.body,
-            map_ranges=map_domain,
-            inputs={**input_mapping, **conn_mapping},
+            sdfg=nsdfg,
+            map_ranges=map_domain or {"__dummy": "0"},
+            inputs=array_mapping,
             outputs=output_mapping,
             symbol_mapping=symbol_mapping,
             schedule=dace.ScheduleType.Sequential,
         )
         access_nodes = {edge.data.data: edge.dst for edge in closure_state.out_edges(map_exit)}
-        transient_to_arg_name_mapping = {
-            memlet.data: name for name, memlet in output_memlets.items()
-        }
         for edge in closure_state.in_edges(map_exit):
             memlet = edge.data
+            if memlet.data not in transient_to_arg_name_mapping:
+                continue
             transient_access = closure_state.add_access(memlet.data)
             closure_state.add_edge(
                 nsdfg_node,
                 edge.src_conn,
                 transient_access,
                 None,
-                dace.Memlet(data=memlet.data, subset="0"),
+                dace.Memlet(data=memlet.data, subset=output_subset),
+            )
+            inner_memlet = dace.Memlet(
+                data=memlet.data, subset=output_subset, other_subset=memlet.subset
             )
-            inner_memlet = dace.Memlet(data=memlet.data, subset="0", other_subset=memlet.subset)
             closure_state.add_edge(transient_access, None, map_exit, edge.dst_conn, inner_memlet)
             closure_state.remove_edge(edge)
-            access_nodes[edge.data.data].data = transient_to_arg_name_mapping[edge.data.data]
+            access_nodes[memlet.data].data = transient_to_arg_name_mapping[memlet.data]
 
         for _, (lb, ub) in closure_domain:
-            map_entry.add_in_connector(lb.value.data)
-            map_entry.add_in_connector(ub.value.data)
-            closure_state.add_edge(
-                lb.value, None, map_entry, lb.value.data, create_memlet_at(lb.value.data, ("0",))
+            for b in lb, ub:
+                if isinstance(b, SymbolExpr):
+                    continue
+                map_entry.add_in_connector(b.value.data)
+                closure_state.add_edge(
+                    b.value,
+                    None,
+                    map_entry,
+                    b.value.data,
+                    create_memlet_at(b.value.data, ("0",)),
+                )
+        return closure_sdfg
+
+    def _visit_scan_stencil_closure(
+        self,
+        node: itir.StencilClosure,
+        array_table: dict[str, dace.data.Array],
+        closure_domain: tuple[tuple[str, tuple[ValueExpr, ValueExpr]], ...],
```

Comment:
Good catch. Shouldn't type-checking have caught it?

Reply:
I don't think so, because you are not doing anything illegal typing-wise. Maybe it shows a warning, but I think mypy only shows warnings when there is an error as well.